tests/specs/session-replay/session-state.e2e.js

```diff
@@ -0,0 +1,131 @@
+
+        await Promise.all([
+          browser.testHandle.expectBlob(),
+          browser.execute(function () { document.querySelector('body').click() })
+        ])
+
+        // session has ended, replay should have set mode to "OFF"
+        const { agentSessions: newSession } = await browser.getAgentSessionInfo()
+        const newSessionClass = Object.values(newSession)[0]
+        expect(newSessionClass.sessionReplay).toEqual(MODE.OFF)
+      })
+    })
+
+    describe('When session resumes', () => {
+      withBrowsersMatching(supportsMultipleTabs)('should take a full snapshot and continue recording', async () => {
+        await browser.url(await browser.testHandle.assetURL('instrumented.html', config()))
+          .then(() => browser.waitForAgentLoad())
+
+        await browser.pause(3000)
+
+        const { events: currentPayload } = await getSR()
+
+        expect(currentPayload.length).toBeGreaterThan(0)
+        // type 2 payloads are snapshots
+        expect(currentPayload.filter(x => x.type === RRWEB_EVENT_TYPES.FullSnapshot).length).toEqual(1)
+
+        const newTab = await browser.createWindow('tab')
+        await browser.switchToWindow(newTab.handle)
+        await browser.url(await browser.testHandle.assetURL('instrumented.html', config()))
+          .then(() => browser.waitForAgentLoad())
+
+        const { events: resumedPayload } = await getSR()
+
+        // payload was harvested, new vis change should trigger a new recording which includes a new full snapshot
+        expect(resumedPayload.length).toBeGreaterThan(0)
+        // type 2 payloads are snapshots
+        expect(resumedPayload.filter(x => x.type === RRWEB_EVENT_TYPES.FullSnapshot).length).toEqual(1)
+
+        await browser.closeWindow()
+        await browser.switchToWindow((await browser.getWindowHandles())[0])
+      })
+    })
+
+    describe('When session pauses', () => {
+      withBrowsersMatching(supportsMultipleTabs)('should pause recording', async () => {
+        await browser.url(await browser.testHandle.assetURL('instrumented.html', config()))
+          .then(() => browser.waitForAgentLoad())
+
+        const newTab = await browser.createWindow('tab')
+        await Promise.all([
+          browser.testHandle.expectBlob(), browser.switchToWindow(newTab.handle), browser.url(await browser.testHandle.assetURL('instrumented.html', { ...config(), loader: 'full' }))])
+        await browser.waitForAgentLoad()
+
+        try {
+        // wait longer than harvest interval
+          await browser.testHandle.expectBlob(6000)
+          // expect bam endpoint to have only been called once (instead of twice)
+          fail('Should not have seen another blob request')
+        } catch (err) {
+          // if expectblob failed, this test passed.
+        } finally {
+          await browser.closeWindow()
+          await browser.switchToWindow((await browser.getWindowHandles())[0])
+        }
```

Comment:
```suggestion
        // Waiting for the second blob should time out, indicating no second call to the BAM endpoint.
        // The wait must be longer than harvest interval.
        await browser.testHandle.expectBlob(6000, true)
        await browser.closeWindow()
        await browser.switchToWindow((await browser.getWindowHandles())[0])
```

Reply:
The purpose of this one is to utilize the expectTimeout flag instead of an expect error construct. It was logging the timeout error even though it was caught.

Reply:
changed